Yoast/wordpress-seo
# Create the indexables page

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* Create an indexables page

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

**Different states of the Indexable page**
* Check the Test Instructions of https://github.com/Yoast/wordpress-seo/pull/18756

**Link Modal**
* Check the Test Instructions of https://github.com/Yoast/wordpress-seo/pull/18747
 
**Repopulating lists**
* Check the Test Instructions of https://github.com/Yoast/wordpress-seo/pull/18735

**Assesments**
* For the `SEO scores` and `Readability scores` lists, confirm that the color of the bullet of each indexable is the same color that appears when going into that post's edit page. Also confirm that no green bullets appear there
* For the `Most linked` list, confirm that posts that are marked as cornerstone content have a star next to their title.

**Error in SEO optimization**
* Check the Test Instructions of https://github.com/Yoast/wordpress-seo/pull/18891

**Error in API calls**
* Check the Test Instructions of https://github.com/Yoast/wordpress-seo/pull/18889

**Ignore and Undo**
* Hide one row in a certain list (List A). Then hide a row in another list (List B).
* Observe the button appearing at the bottom to undo the hiding.
* Now in List B, click the three-dot-menu, and click "restore hidden items". The aforementioned button should disappear and the row you've hidden should come back. List A should be unaffected.
* Now, again, hide a row in List A. Then hide a row in List B.
* Observe the button appearing at the bottom to undo the hiding.
* Now, in List A, click the three dot menu and click restore hidden items. The button should not disappear. Clicking it should still work. The row you've hidden in List A should be back.
* Hide various rows in all lists.
* Click the  "Restore all hidden items" button. The lists should reload (but not the whole page).

**Shortlinks**
* Check the Test Instructions of https://github.com/Yoast/wordpress-seo/pull/18902


**Tracking state test instructions**
* Make sure you have a clean datatabase (start with a fresh installation of Yoast SEO Free or  use the `Reset Options` functionality in Yoast Test Helper)
* Visit the indexables overview page, you should see the `No data to show yet` page
* In the database, go to the `PREFIX_options` table and copy the value field of the `wpseo` option
* Use a tool such as [https://www.unserialize.com/](https://www.unserialize.com/) to unserialize the string
* Look for the `indexables_overview_state` key, its value should be `optimization-not-complete`
* Add the following filter to your site:
```
add_filter( 'wpseo_posts_threshold', 'custom_posts_threshold', 1 );
function custom_posts_threshold() {
  return 2000;
 }
```
Posts and pages without a focus keyphrase
* Go back to the indexables overview page and click on `Start SEO data optimization`
* Once the process is finished, you'll be redirected to the `Start writing content!` page 
* Refresh the dabase view and check the value of the `indexables_overview_state`: it should now be set to `not-enough-content`
Remove the `custom_posts_threshold` filter you have added before and add the following one:
```
add_filter( 'wpseo_analyzed_posts_threshold', 'custom_analyzed_posts_threshold', 1 );
function custom_analyzed_posts_threshold() {
  return 1.0;
 }
```
* Refresh the indexables overview page
* Refresh the dabase view and check the value of the `indexables_overview_state`: it should now be set to `not-enough-analysed-content`
* Remove the `custom_analyzed_posts_threshold` filter
* Refresh the indexables overview page
* Refresh the dabase view and check the value of the `indexables_overview_state`: it should now be set to `all-good`


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes #
